### DISCUSSION TITLE
Ender3 V2/S1 Professional Firmware

### DISCUSSION THREAD

**Discussion by mriscoc** [mriscoc/Ender3V2S1#13] [Announcements]
Universal New Year Edition!
============

This firmware is based on Marlin, has been optimized and has enhancements for the Creality Ender 3 V2/S1 printer. This is an **universal firmware** that can be used in both RET6 and RCT6 SoC chips that have
256k or 512k Flash memory. 

## Versions
- **Ender 3v2 Manual Mesh**: if you have the Ender 3v2  near to stock then use this version.
- **Ender 3v2 BLTouch**: use this version if you installed a CR/3D/BLTouch level sensor in the Ender 3v2.
- **Ender 3S1**: use this version if you have a Ender 3S1 printer

<img align="left" src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main.jpg" width="270" height="480" />
<img  src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main-iconset.jpg" width="270" height="480" />

The binary files of this release are up to date and you can download them from the assets at the bottom of this page. To know if you have the last release go to the info page in the display of your printer and compare the release build date with the name of the binary file. The filename follows the convention:  
  
Ender3v2-422-ManualMesh-**YYYYMMDD**.bin
  
Where YYYYMMDD indicates the date of the build. 422 refers to the mainboard version 4.2.2, and 427 is for the 4.2.7 mainboard version. The **Volc** versions have support for the **E3D V6 Volcano** hot ends and any heat block that have a 05-class temperature sensor a 100kO ATC Semitec 104GT-2/104NT-4-R025H42G thermistor (as the Hemera).

<img src="https://cdn.shopify.com/s/files/1/0046/3781/8929/files/1-11.jpg" width="320" height="320" />

## Community links
* Telegram
* [Reddit](https://www.reddit.com/r/Ender3v2Firmware) 
* [Ender 3V2 Facebook](https://www.facebook.com/groups/ender3v2firmware)
* [Ender 3S1 Facebook](https://www.facebook.com/groups/ender3s1printer)

## Donations
Thank you for your support, I receive donations through [Patreon](https://www.patreon.com/mriscoc) and [Paypal](https://www.paypal.com/paypalme/mriscoc)   

[<img src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif">](https://www.paypal.com/donate?business=85SPAAR6UZEE8&currency_code=USD)

# Firmware features

* Improved support for Manual Mesh and CR/3D/BLTouch
* 4.2.2, 4.2.7 Boards and V6 Volcano hotends (300?C) supportt
* User configurable power lost recovery enabler
* Bed Tramming assistant 
* Long filename support with scroll
* Baby Z-Stepping
* Firmware Retract
* Advanced pause feature
* Configurable park-head position (default x=240 y=220)
* Filament change support
* Filament management (load/unload/park head) in prepare menu
* Print Flow on Motion and Tune menu 
* Home axis after motor disabling
* Fastest Z feed rate
* Live move for all motors but extruder
* Support for S-curve Acceleration
* Reboot printer item on control menu
* Increase extruder feed rate limit to 150 for BMG users
* Allows to invert direction of the extruder on Motion menu
* Selectable runout filament sensor active state (LOW/HIGH/MOTION)
* Support for BTT Smart filament sensor
* Editable runout distance
* Support for ?SD extensions
* PID popup with temperature plot
* Lock screen
* Editable physical dimensions
* Mesh viewer
* Assisted Bed Tramming
* Custom pre-heat
* Enable/disable beeps sound
* New print g-code preview (use [this ](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview)guide)
* New print done screen
* Two decimal positions for axis coordinates value
* New Z-Offset wizard
* New PID settings menus
* PID moves the head to the center for more accuracy
* Leveling bed temperature linked to bed PID temperature
* Editable extrude minimum temperature
* Adds brightness settings on tune menu
* New status line system, allows the scroll of large messages
* Home submenu with individual axis homing
* Enabled user setting for 115200 bauds (Creality Box compatibility)
* Head parking before recovery after power failure
* Offset icon change to show leveling status
* Live end-stop diagnostic page
* Move bed to front after cancel a print.
* Editable firmware retracts settings for Tune and filament settings menu
* Print Statistics page accessible from the Advanced Settings menu

![latest release](https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/New-beta.gif)

### Manual mesh
* Assisted manual mesh leveling menu
* Selectable mesh points from 2x2 to 9x9
* Mesh viewer

### CR/3D/BLTouch
* Safe Probe Z-Homing
* Bed auto level
* High speed probe
* Retry bed level on error
* Mesh viewer
* Assisted tramming
* Selectable mesh points from 2x2 to 9x9
* Allows to the user to cancel an auto leveling process 
* Increase Max Probe Offset and Probing Margin to 60 mm
* M48 Probe Test to Probe Settings menu
* Probe Stow/Deploy in Probe Settings menu
* Option to disable HS mode for better compatibility with CRTouch
* Option to change the Z Probe feed rate

>Use the 2x2 mesh for first level you bed!

<a href="https://youtu.be/qzYJC2AHbW8"><img src="https://img.youtube.com/vi/qzYJC2AHbW8/maxresdefault.jpg" width="640" height="400" /></a>  
  
### G-Codes enhanced support
* Filament Load Unload G-codes (M701/M702)
* Support for M600 filament change
* Support for G-code M117 messages
* Enable M575 change baud rate by G-code
* Extended M75 with filename support, use M75 {{ event.name }} in g-code script
* Enable ARC and 'P' parameter to specify complete circles (G2/G3)
* G35 support
* M155 Geometry report
* Native support of marlin brightness and M256 g-code

### Support for special nonstandard Creality C G-Codes
* Implemented C11 En Rn Gn Bn gcode for colorize UI elements
* Added enhanced C12 g-code to get the bed z-value of a given X, Y point with a probe
* New g-code C108 to cancel screens waiting for user simple confirmation
* New g-code C29 N# to set amount of grid points of the mesh leveling
* New g-code C101 X### Y### z### to setup physical limits
* New g-code C102 X### Y### to setup bed size
* New g-code C125 X### Y### z### to setup the park position
* New g-code C412 M# set run-out sensor active state (M0:LOW, M1:HIGH, M2:MOTION)
* New g-code C510 to lock/unlock the screen (C510 U1 to unlock)
* New g-code C562 E# to invert the Extruder (E1 to invert)
* New g-code C851 M### F#### H# to set probing margin, Z feed rate and HS mode of the probe

<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/themes/Select_Colors.jpg" width="270" height="480" /> 

### Advanced settings menu
* Home XYZ offsets menu
* Probe offsets menu (BLTouch version)
* Customizable park-head position
* Customizable bed size and z heigh up to 999 x 999 x 999
* Launch PID for nozzle and bed with graphical feedback
* Filament run-out sensor with active state setup (LOW/HIGH)
* Filament load and unload lengths menu items in run-out menu
* Power-loss recovery enable/disable option
* LCD Brightness control
* Select Color Sub Menu
* Beep/Sound enable/disable option
* Selectable mesh point (BLTouch version)
* Mesh viewer
* Lock screen

### Special Octoprint / Host print support
* High speed (250000 baud) serial communication
* In screen control of host printing (M73, M75-M77) 
* Display messages on status line (M117)
* Color change of UI elements for visual status report (C11)
* Binary file transfer for remote firmware upgrade
* List binary (.bin) files (M20) for flash firmware support with BFT
* Arc Welder Octoprint plugin support
* EEPROM Editor Octoprint plugin support
* Host action commands for filament runout
* Emergency Parser active
* Advanced OK

**Octoprint bed visualizer support**
<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/Bed-visualizer.png" width="350" height="225" />

### Display enhancements
* Extended DWIN library
* Enhanced status messages
* Improved status bar with temperature, flow, fan, z-offset and xyz positions
* PID auto tune progress on display with plot
* Leveling progress on display
* Boot screen for M997
* Transparent icons for better color theme adaptability
* New print g-code preview
* New print done screen

**A lot of additions and fixes from Bugfix development branch**

## More info on the wiki pages ##
- [**How to install the firmware**](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-install-the-firmware)

- [**Octoprint settings**](https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint)

- [**How to make the Manual Mesh**](https://github.com/mriscoc/Ender3V2S1/wiki/Manual-Mesh)

- [**How to install a 3D/BLTouch**](https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch)

- [**How to apply a color theme**](https://github.com/mriscoc/Ender3V2S1/wiki/Color-Themes)

## Remarks ##
Please remember that it is a preliminary and bugfix release be careful. If you are an advanced user, you can get the last **experimental** version at the Telegram firmware group: https://t.me/Ender3v2_Firmware where we test the firmware and has new features added. When this version gets tested enough is renamed to **BETA** and is available to be tested by anyone, at this time, the previous BETA is released as Ender3v2. For compile yourself, grab the current base [source files](https://github.com/mriscoc/Ender3V2S1) then apply the correct configuration files from the configurations folder, most of the more recent features or additions will be released through pull requests to the official Marlin repository.
 
## What's new in this release ##
* Applied official Bug-fix branch of Marlin
* Universal RCT6/RET6 Ender 3 V2/S1 Support

https://user-images.githubusercontent.com/2745567/143180943-f69f5967-34c1-421f-86a1-466ecccab97d.mp4

<hr /><em>This discussion was created from the release <a href='https://github.com/mriscoc/Ender3V2S1/releases/tag/20220202'>Ender3 V2/S1 Professional Firmware</a>.</em>

**Comment by josederas** [comment]
Could you please clarify what the first 2 LCD screenshots depict?  They have different fonts and icons, does that mean that's selectable?

**Reply by mriscoc** [reply]
You can use the stock icon assets or the Giadej icon assets from here: https://github.com/mriscoc/Ender3V2S1/wiki/How-to-update-the-display
Only compatible with the original Ender3v2 display.

**Reply by josederas** [reply]
Great thanks.  To be clear, the fonts used also change?  Also, Giadej doesn't seem to match the picture, I assume the picture on the link is correct but not the above picture?

**Comment by Dooti-2** [comment]
This is amazing.  I have searched for SO long for the right firmware...have compiled my own, made many tweaks but still couldnt get one just right for my printer with my mods.  My favorite feature is the fact that I can invert E from the printer -- mind blown -- thank you so much.

**Comment by grrmisfit** [comment]
mesh isnt saving, did auto build mesh , it runs, but when i click view mesh in advanced it says no valid mesh

**Reply by mriscoc** [reply]
For preserving the mesh after reboot you must to save it going to control/store settings.

**Comment by stanleyyyy** [comment]
Hi,
I tried to compile the binary for my Ender 3 S1, using the configuration from configurations/Ender3 S1 (I copied it to Marlin folder) and environment STM32F103RE in Vscode/Platformtio.
The binary kind of worked but the print got always interrupted because of filament runaway sensor failure. Can you please confirm if the configuration in the github is the same one which you are using to build your S1 binaries?
Also I have noticed that X offset on my S1 is outside of bed area, so the first "line" which is used to flush the extruder is printed outside of the bed. Just reporting :)
Anyway, this seems to be a very promising project and I really appreciate your effort! Cheers!

**Reply by mriscoc** [reply]
Hi for the Ender3S1 the environment must be STM32F103RC_creality (256K). Could you verify in filament settings if the runout active is in HIGH state? If not, please change to HIGH and store settings in Control menu.

**Comment by stanleyyyy** [comment]
Hi,
yes, in the meantime I figured that out.. that value is reset during
factory reset operation to 0 which basically inverts the behavior and I
have to manually
set the switch every time I do an update or a factory reset..

So I decided to make a patch to solve this problem by explicitly
configuring the value in DWIN_SetDataDefaults() - see attachment.
Cheers,
Stanislav



ne 6. 2. 2022 o 17:16 Miguel Risco-Castillo ***@***.***>
napísal(a):

> Hi for the Ender3S1 the environment must be STM32F103RC_creality (256K).
> Could you verify in filament settings if the runout active is in HIGH state?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mriscoc/Ender3V2S1/discussions/13#discussioncomment-2120432>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAITZ25UOTPH3TQGGFAUARTUZ2NGTANCNFSM5NNWBGOQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
S pozdravom,
Ružáni

Ing. Stanislav Ruzani, skype: stanleyy

**Comment by curiousmongrel** [comment]
I cannot find the 'Volc' versions (i.e. "The Volc versions have support for the E3D V6 Volcano hot ends...").  Am I missing something obvious or are they not there?? TIA

**Reply by mriscoc** [reply]
To avoid confusion with the new users I'm thinking of moving the special compiles to another repository. If you do not compile your own version please wait, I will publish maybe here: https://github.com/mriscoc/Marlin_Configurations

**Reply by Urmoms2a** [reply]
I have you old version of the 427 manual mesh build for volc you mad back in October I believe after I did a request you changed the temp for me I appreciate it . I would like to use this new build and have a request for the volc to be 315 so the print temp can go to 300 . My version that you built is 300 and print temp is 285 so the 15 degree throws it off . 315 would be awesome ! Thank you!

**Reply by mriscoc** [reply]
That 15°C is an overshoot for security, when you do a PID for example, the temperature raises over:  `HEATER_0_MAXTEMP - HOTEND_OVERSHOOT`
Have you a link where indicates what is the maximum temperature that the V6/Volcano can support?

**Reply by mriscoc** [reply]
Volcano Builds
https://github.com/mriscoc/Marlin_Configurations/releases/tag/20220209

**Reply by Urmoms2a** [reply]
https://wiki.e3d-online.com/FAQs#:~:text=v6%20Specific%20Questions-,Max%20Temp,C)%20with%20a%20thermocouple%20upgrade.

**Reply by mriscoc** [reply]
That is not an original V6/Volcano but an upgraded one, then you must select the appropriate thermal sensor, in that case a thermocouple. This configuration must have a different name, could you specify what is the class of thermal sensor to put in TEMP_SENSOR_0?

**Reply by Urmoms2a** [reply]
Here is a link to thermocouple that would be used in 315c application for hotend .

**Reply by Urmoms2a** [reply]
https://www.matterhackers.com/store/printer-accessories/e3d-type-k-thermocouple-cartridge

**Reply by Urmoms2a** [reply]
I enjoy your build by far better than anyone else and I tell everyone about it and send them links to yours . If you could do a 427 manualesh volc build that I could have the actual print temp at 315 that would be perfect and all of us that print in cf pollycarbonate for testing part would be amazing . Thank you !

**Reply by mriscoc** [reply]
I could but that temperature could be risky because it is outside of the temperature range of the V6/Volcano thermistor. 315 °C means that if you do a PID, the máximum temperature could reach the 330°C and the manufacturer did set the maximum to 295°C. I can do a build for a higher temperature but the thermistor must be changed to a thermocouple and in this case, it will not be a standard E3D V6/Volcano, the exact thermocouple must be defined in TEMP_SENSOR_0.

**Comment by Urmoms2a** [comment]
Looks like they say 295 per thermistor ..so if you did a "hi temp" build for us with super modded printers that print polycarbonate and ezpcf it would help if the actual hot end went to 300 ..so would have to be 315 in the script .

**Reply by mriscoc** [reply]
Please, give me the correct TEMP_SENSOR_0 to change the configuration files and I will do a build for it.

**Reply by Urmoms2a** [reply]
TEMP_SENSOR_0 20

20    for pt100 capable of 500c for e3d v6

**Reply by mriscoc** [reply]
Ok, I did release a special version for that request you can find here: https://github.com/mriscoc/Marlin_Configurations/releases/tag/20220214-AMT20 but for future changes is better that you try to compile by yourself using the generated configuration files.

**Comment by johnmorenov** [comment]
Hi, I have a Ender 3 S1 Pro, I know that pro firmware is not compatible with touchscreen, but if I change that screen to one with a rotary knoob (V2/S1), that way I'd think if it could be installed, but, how I can change max temperature for hotend (300°C) and bed (110°C)? thanks

**Comment by mgadei** [comment]
Hi, I would like to have precompiled version LA , for ender 3 v2 with sprite pro, and bltouch. You created one for T5 but none for T13. Please

**Reply by mriscoc** [reply]
The precompiled Special Versions are sponsored versions, but you can compile your own using the configurator Python script. https://github.com/mriscoc/Special_Configurations

**Comment by mgadei** [comment]
Hi Miguel,What I need is a linear advance version for Ender 3 v2 with 4.2.2 board , Sprite extruder Pro and CRtouch. I will sponsor you for this version, after sending my email. I am not good at all in compiling, python and really don’t want to mess with this now. I have too many things to learn for my work and the bowl is full.Thank you in advanceMihaiOn Nov 9, 2022, at 23:54, Miguel Risco-Castillo ***@***.***> wrote:﻿
The precompiled Special Versions are sponsored versions, but you can compile your own using the configurator Python script. https://github.com/mriscoc/Special_Configurations

**Reply by mriscoc** [reply]
HI, Thank you, your sponsored version is here: https://github.com/mriscoc/Special_Configurations/releases/download/LinearAdvance/Ender3V2-422-BLTUBL-T13-LA-20221113.bin

**Comment by TheRedneckedGinger** [comment]
Hello,
I have an Ender 3 v2 (board 4.2.2) with a sprite Extruder that I upgraded to. I’ve just purchased a BL touch (V3.1), have it installed but when I hit ‘Auto-home’ the head goes to the middle of the bed and doesn’t home the z-axis. When I try to auto level the bed, it starts probing but then X-axis tries to go to maximum and the motor starts skipping on the belt.
Would you be able to help me with this issue?
Oh, the firmware I’m using is miscroc’s T13 BLTUBL that I downloaded from here. 
thanks for your help

**Reply by TheRedneckedGinger** [reply]
Ah ha, nevermind. I just forgot to change the probe offset position.
Just wanna say thank you for your contributions to the Ender community.

**Comment by aBugMansLife** [comment]
Hello,
Is there a way to show post print information?
Actual printing time and filament used?
Currently I only get the thumbnail of the print after completion without any other information.

**Reply by mriscoc** [reply]
Yes, you can change `Draw_PrintDone` code.  Currently Thumbnail and print time are displayed.

**Reply by aBugMansLife** [reply]
Coolness! Thank you very very much!
